torch/csrc/lazy/tutorial.md

```diff
@@ -0,0 +1,251 @@
+
+TODO: Spiel about awesomeness of whole graph compilations 
+
+However, if you model is wildly dynamic and contains loops that always run different number of times or if statements one after another that explode into different traces every time you run your model, the backend will spend non-trivial amount of time compiling each trace even though the latter is used only for a few times. 
+
+Alright, at this point, you should have learned the main ideas behind Lazy Tensor, most common usage patterns and APIs.
+Also, you are hopefully as inspired and motivated about Lazy Tensor as I am.
+
+Let's see now how we can run a full training loop with an optimizer and backward pass! We will learn a few more important concepts and APIs.
+
+## MNIST MLP
+
+We will adapt the following example running MNIST_MLP from [pytorch/examples](https://github.com/pytorch/examples/blob/main/mnist/main.py)
+
+First, we need to install one single dependency, `torchvision`
+
+```
+pip install torchvision
+```
+
+`torchvision` comes with MNIST dataset w/ images of handwritten digits, which we will be using for training.
+
+Here's our model definition:
+
+```python
+class Net(nn.Module):
+    def __init__(self):
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 32, 3, 1)
+        self.conv2 = nn.Conv2d(32, 64, 3, 1)
+        self.dropout1 = nn.Dropout(0.25)
+        self.dropout2 = nn.Dropout(0.5)
+        self.fc1 = nn.Linear(9216, 128)
+        self.fc2 = nn.Linear(128, 10)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = F.relu(x)
+        x = self.conv2(x)
+        x = F.relu(x)
+        x = F.max_pool2d(x, 2)
+        x = self.dropout1(x)
+        x = torch.flatten(x, 1)
+        x = self.fc1(x)
+        x = F.relu(x)
+        x = self.dropout2(x)
+        x = self.fc2(x)
+        output = F.log_softmax(x, dim=1)
+        return output
+```
+
+We are using a multi-level perceptron model with two convolutions, two linear layers and activations sandwiched in between.
+
+Let's set up a loader that would feed the `MNIST` dataset in `train` to our model.
+We are going to run the training loop for 14 epochs which is what the original MNIST example uses.
+**Note, we had to move the model to the Lazy device, `Net().to(device)`. This is very similar to what we would have done had we been training this model on a GPU.**
+
+The rest of the code is pretty standard boilerplate.
+
+```python
+import os
+import torch
+import torchvision
+
+if __name__='__main__':
+    bsz = 64
+    device = 'lazy'
+    epochs = 14
+    log_interval = 10
+    train_kwargs = {'batch_size': args.batch_size}
+    test_kwargs = {'batch_size': args.test_batch_size}
+    # if we want to use CUDA
+    if "LTC_TS_CUDA" in os.environ:
+        train_kwargs = {'num_workers': 1,
```

Comment:
cuda_kwargs?

Reply:
Yup, last night I made sure the model runs and trains, I'll push the fixes shortly